lib/lower/convert-expr.cc

```diff
@@ -280,7 +280,31 @@ class ExprLowering {
   M::Value genval(Ev::NullPointer const &) { TODO(); }
   M::Value genval(Ev::StructureConstructor const &) { TODO(); }
   M::Value genval(Ev::ImpliedDoIndex const &) { TODO(); }
-  M::Value genval(Ev::DescriptorInquiry const &) { TODO(); }
+  M::Value genval(Ev::DescriptorInquiry const &desc) {
+    auto descRef{symMap.lookupSymbol(desc.base().GetLastSymbol())};
+    assert(descRef && "no mlir::Value associated to Symbol");
+    auto descType{descRef->getType()};
+    M::Value res{};
+    switch (desc.field()) {
+    case Ev::DescriptorInquiry::Field::Len:
+      if (auto boxCharType{descType.dyn_cast<fir::BoxCharType>()}) {
+        auto refType{fir::ReferenceType::get(boxCharType.getEleTy())};
+        auto lenType{mlir::IntegerType::get(64, builder.getContext())};
+        res = builder
+                  .create<fir::UnboxCharOp>(getLoc(), refType, lenType, descRef)
+                  .getResult(1);
+      } else if (descType.isa<fir::BoxType>()) {
+        TODO();
+      } else {
+        assert(false && "not a descriptor");
```

Comment:
I started converting some of these "impossible cases" from asserts to mlir::emitError() calls.  Although the asserts are nice when running under a debugger and when the code actually does the impossible.

Reply:
Ok, I missed that change, I think convert-expr.cc is not yet using `mlir::emitError()` do you want me to change this one already or leave it and change it with all convert-expr.cc asserts later ?

Reply:
You can leave it. It was more of a heads-up.